core/src/main/java/software/amazon/awssdk/async/AsyncResponseHandler.java

```diff
@@ -46,6 +49,8 @@
      * will be treated as a failure of the response and the {@link #exceptionOccurred(Throwable)} callback will be invoked.
      * </p>
      *
+     * <p>This callback may never be called if the response has no content or if an error occurs.</p>
```

Comment:
Yes, it will use https://github.com/aws/aws-sdk-java-v2/blob/master/core/src/main/java/software/amazon/awssdk/http/async/SyncResponseHandlerAdapter.java for non-streaming content. From the HTTP layer perspective there is no such thing as streaming vs non-streaming. There's just content it delivers asynchronously to the response handler.